File: SuperiorPackGroup/BLL/PackingListReportBLL.vb
Imports SuperiorPackGroup.SPGReportsTableAdapters

Public Class PackingListReportBLL

    Private m_PackingListReportTableAdapter As PackingListReportTableAdapter = Nothing

    Public ReadOnly Property Adapter() As PackingListReportTableAdapter
        Get
            If (m_PackingListReportTableAdapter Is Nothing) Then
                m_PackingListReportTableAdapter = New PackingListReportTableAdapter
                m_PackingListReportTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return m_PackingListReportTableAdapter
        End Get
    End Property

    Public Function GetPackingListreport(ByVal ShippingID As Integer) As SPGReports.PackingLIstReportDataTable

        Try
            Return Adapter.GetPackingList(ShippingID)
        Catch ex As Exception
            MessageBox.Show(String.Format("There was an error while retrieving the information to print{0}{1}", vbCrLf, ex.Message))
            Return Nothing
        End Try

    End Function

End Class

File: SuperiorPackGroup/BLL/PackingListReportBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGReportsTableAdapters;

namespace SuperiorPackGroup
{
	public class PackingListReportBLL
	{

		private PackingListReportTableAdapter m_PackingListReportTableAdapter = null;

		public PackingListReportTableAdapter Adapter
		{
			get
			{
				if (m_PackingListReportTableAdapter == null)
				{
					m_PackingListReportTableAdapter = new PackingListReportTableAdapter();
					m_PackingListReportTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return m_PackingListReportTableAdapter;
			}
		}

		public SPGReports.PackingListReportDataTable GetPackingListreport(int ShippingID)
		{

			try
			{
				return Adapter.GetPackingList(ShippingID);
			}
			catch (Exception ex)
			{
				MessageBox.Show(string.Format("There was an error while retrieving the information to print{0}{1}", Environment.NewLine, ex.Message));
				return null;
			}

		}

	}

}